src/core/ext/transport/binder/wire_format/wire_reader_impl.h

```diff
@@ -99,9 +101,11 @@ class WireReaderImpl : public WireReader {
  private:
   absl::Status ProcessStreamingTransaction(transaction_code_t code,
                                            ReadableParcel* parcel);
-  absl::Status ProcessStreamingTransactionImpl(transaction_code_t code,
-                                               ReadableParcel* parcel,
-                                               int* cancellation_flags)
+  absl::Status ProcessStreamingTransactionImpl(
+      transaction_code_t code, ReadableParcel* parcel, int* cancellation_flags,
+      // The queue saves the actions needed to be done "WITHOUT" `mu_`.
+      // It prevents deadlock against wire writer issues.
+      std::queue<std::function<void()>>& deferred_func_queue)
```

Comment:
Not sure if it will be cleaner, but it is possible to combine multiple callbacks in that function so it can return single callback instead of a queue of callbacks.

Reply:
I personally think multiple callbacks in the queue will be easier to understand/maintain for this case.